packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -569,6 +659,57 @@ abstract class TableBase extends Resource implements ITable {
   }
 
   protected abstract get hasIndex(): boolean;
+
+  /**
+   * Adds an IAM policy statement associated with this table to an IAM
+   * principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param tableActions The set of actions to allow (i.e. "dynamodb:PutItem", "dynamodb:GetItem", ...)
+   * @param keyActions The set of actions to allow (i.e. "kms:Encrypt", "kms:Decrypt", ...)
+   */
+  private tableGrantKeyActions(grantee: iam.IGrantable, tableActions: string[], keyActions?: string[]) {
```

Comment:
I'd suggest changing the signature for improved readability on the calling sites, while also combining it with the Stream form of it:

```suggestion
  private combinedGrant(
    grantee: iam.IGrantable,
    opts: { keyActions?: string[], streamActions?: string[], tableActions?: string[] }
  ) {
```

That'll effectively remove some duplicated code, and make the method more polyvalent.